build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -4,294 +4,187 @@
-					// Generate and write the modified manifest
-					const manifestFile = await generateReleaseReportForUnreleasedVersions(
-						GITHUB_RELEASE_URL,
-						devVersion,
-						logger,
-					);
-					if (manifestFile !== undefined) {
-						await writeManifestToFile(manifestFile, flags.output, logger);
-					}
-				}
-			} else if (buildNumber === undefined) {
-				this.log(
-					`No successful build found for ${flags.sourceBranch} branch in the last 24 hours`,
-				);
-			}
+			this.log("Files processed successfully.");
 		} catch (error: unknown) {
-			throw new Error(`Error creating manifest file: ${error}`);
+			this.error(`Unable to process manifest files: ${error}`);
 		}
 	}
 }
 
 /**
- * Fetches the first successful build number in the last 24 hours.
- * @param authHeader - Auth Header
- * @param adoUrl - Azure DevOps API URL
- * @param sourceBranch - Source branch name
- * @returns The build number if successful, otherwise undefined.
+ * Generates release reports for unreleased versions based on specified manifest files.
+ * @param caretManifestFilePath - The path to the caret manifest file.
+ * @param simpleManifestFilePath - The path to the simple manifest file.
+ * @param version - The version string for the reports.
+ * @param outDir - The output directory for the reports.
+ * @param logger - The logger object for logging messages.
  */
-async function getFirstSuccessfulBuild(
-	authHeader: string,
-	adoUrl: string,
-	sourceBranch: string,
-	log?: Logger,
-): Promise<string | undefined> {
-	try {
-		const response = await fetch(adoUrl, { headers: { Authorization: authHeader } });
-
-		/* eslint-disable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call */
-		const data = await response.json();
-
-		const twentyFourHoursAgo = new Date();
-		twentyFourHoursAgo.setHours(twentyFourHoursAgo.getHours() - 24);
-
-		// Filter builds by date first
-		const recentBuilds = data.value.filter(
-			(build: IBuildDetails) => new Date(build.finishTime) >= twentyFourHoursAgo,
-		);
-
-		if (recentBuilds === undefined || recentBuilds.length === 0) {
-			log?.errorLog(`No successful builds found in the last 24 hours`);
-		}
-
-		const successfulBuilds = recentBuilds.filter(
-			(build: IBuildDetails) =>
-				build.definition.name === "Build - client packages" &&
-				build.status === "completed" &&
-				build.result === "succeeded" &&
-				build.sourceBranch === `refs/heads/${sourceBranch}`,
-		);
-
-		return successfulBuilds.length > 0
-			? (successfulBuilds[0].buildNumber as string)
-			: undefined;
-		/* eslint-enable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call  */
-	} catch (error) {
-		log?.errorLog("Error fetching successful builds:", error);
-		return undefined;
+async function generateReleaseReportForUnreleasedVersions(
+	caretManifestFilePath: string,
+	simpleManifestFilePath: string,
```

Comment:
I've been thinking about this a bit, and the `full` report version has everything that the other reports do - why not take the full report in as input and transform it as needed? You can read in the JSON file cast it as a ReleaseReport type and then transform as needed. See toReportKind in library/release.ts too.